package/MDAnalysis/core/universe.py

```diff
@@ -465,7 +465,7 @@ def transfer_to_memory(self, start=None, stop=None, step=None, verbose=None, qui
             # fall back to a slower approach
             except AttributeError:
                 pm = ProgressMeter(self.trajectory.n_frames,
```

Comment:
the iteration might have less frames if  any of `start`,`stop`,`step` deviate from their default values